Find mxDateTime.h automatically

cthart opened the pull request:
We use mxDateTime in our old codebase. Right now the only way to compile this is by editing setup.cfg
This patch lets setup.py find the mxDateTime headers automatically.

dvarrazzo commented:
Thank you for this. So do you prefer automatic search over specifying a build option? If that's good for you it's ok for me.

I've squashed your changes and added a line into the news file. Once tests pass I'll merge it.

cthart commented:
Hi Daniele,

Sorry for my tardy response, I've been working on some other things the
last few days.

Thanks for applying my patch!

Well, you and I were both after the simplest way/patch that would work.
When I first started investigating this issue everything pointed to needing
to provide a parameter to pip when building. But the more I dug into it the
more apparent it became that the existing setup.py was simply not looking
in all possible places.

I tested with

pip install /home/colin/git/psycopg2

and then running

python -c "import psycopg2;
psycopg2.extensions.register_type(psycopg2._psycopg.MXDATETIME); print
'OK'";

(egenix-mx-base previously installed via pip of course)

So I'm confident this will also work when installed in the standard way
with "pip install psycopg"

When do you think the next release will be made? I'd like to start setting
up a new production server without any "hacks" to get it installed.

NB mxDateTime works with Python 3 so I don't expect it to go away anytime
soon, especially for people with large legacy codebases like ours. We also
have quite some services that use xmlrpc which pickles the data going
across, so you get the same data types on both sides. This makes it pretty
difficult to change to a regular datetime as then we would have to change
both sides of the service at the same time, potentially requiring us to
update as many as 20 locations... Oh well, fact of life, I guess :-)

Thanks again!

/Colin



On Wed, 6 May 2020 at 04:51, Daniele Varrazzo <notifications@github.com>
wrote:

> Thank you for this. So do you prefer automatic search over specifying a
> build option? If that's good for you it's ok for me.
>
> I've squashed your changes and added a line into the news file. Once tests
> pass I'll merge it.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/psycopg/psycopg2/pull/1102#issuecomment-624415205>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJHP7I43GYRC4OYC25YA6TRQDGBTANCNFSM4MZAJGKA>
> .
>

dvarrazzo commented:
> I tested with
>
> pip install /home/colin/git/psycopg2 

I guess that's good enough. What I had in mind is:

    pip install -e git+https://github.com/psycopg/psycopg2.git#egg=psycopg2

but if you merged from upstream in your working copy that's good.

> NB mxDateTime works with Python 3 so I don't expect it to go away anytime
> soon, especially for people with large legacy codebases like ours.

Well, thank you for your input in suggesting how to make it more usable! I only used mx before pip existed: I didn't even had idea that the include files were installed via pip. 

I would like to make a new release in a couple of weeks.

Please consider sponsoring the project: see the sponsorship options on top of the page.
